datadog/fwprovider/resource_datadog_integration_aws_event_bridge.go

```diff
@@ -0,0 +1,257 @@
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"create_event_bus": schema.BoolAttribute{
+				Computed:    true,
+				Default:     booldefault.StaticBool(true),
+				Description: "True if Datadog should create the event bus in addition to the event source. Requires the `events:CreateEventBus` permission.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
+			},
+			"event_generator_name": schema.StringAttribute{
+				Description: "The given part of the event source name, which is then combined with an assigned suffix to form the full name.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"region": schema.StringAttribute{
+				Description: "The event source's [AWS region](https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints).",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"id": utils.ResourceIDAttribute(),
+		},
+	}
+}
+
+func (r *integrationAwsEventBridgeResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
+}
+
+func (r *integrationAwsEventBridgeResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state integrationAwsEventBridgeModel
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	resp, httpResp, err := r.Api.ListAWSEventBridgeSources(r.Auth)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "Error listing AWS EventBridge Event Sources"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+
+	found := false
+	matchedEventHub := integrationAwsEventBridgeModel{}
+
+	if accounts, ok := resp.GetAccountsOk(); ok && len(*accounts) > 0 {
+		for _, account := range *accounts {
+			if found {
+				break
+			}
+			if eventhubs, ok := account.GetEventHubsOk(); ok && len(*eventhubs) > 0 {
+				for _, eventhub := range *eventhubs {
+					if *eventhub.Name == state.ID.ValueString() {
+						matchedEventHub.ID = types.StringValue(*eventhub.Name)
+						matchedEventHub.AccountId = types.StringValue(account.GetAccountId())
+						matchedEventHub.Region = types.StringValue(eventhub.GetRegion())
+						found = true
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if !found {
+		response.State.RemoveResource(ctx)
+		response.Diagnostics.AddError("Import error", "No matching Event Source found")
```

Comment:
```suggestion
		response.Diagnostics.AddWarning("Read error", "No matching Event Source found.")
```
We shouldn't error on a missing event source. Terraform calls the read function during planning, so erroring here will softlock the resource and prevent terraform from recreating the event source to fix itself.